src/main/webapp/app/exercises/shared/result/result.component.ts

```diff
@@ -86,10 +95,7 @@ export class ResultComponent implements OnInit, OnChanges {
     @Input() showUngradedResults: boolean;
     @Input() showGradedBadge = false;
     @Input() showTestDetails = false;
-    /**
-     * Information about a missing result can be set to communicate problems and give hints how to respond.
-     */
-    @Input() missingResultInfo?: { message: string; tooltip: string } = undefined;
+    @Input() missingResultInfo?: MissingResultInfo = undefined;
```

Comment:
Probably last thing :D

I'd like to question the default value. I think when using an enum where you need a base/default value like in this case, that you don't take such thing as `undefined` or `null` because that's wrong. The result info is not undefined but none.
So I think it would be a better style to add a third enum option and taking this a the default value.

Do you agree there?